pkg/controller/infinispan/resources/operatorconfig/operatorconfig_controller.go

```diff
@@ -0,0 +1,127 @@
+package operatorconfig
+
+import (
+	"context"
+	"strings"
+
+	"github.com/go-logr/logr"
+	kube "github.com/infinispan/infinispan-operator/pkg/kubernetes"
+	"github.com/operator-framework/operator-sdk/pkg/k8sutil"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	ControllerName = "operator-config-controller"
+	configMapName  = "infinispan-operator-config"
```

Comment:
I think we need to append the operator major version or channel to this. In the model where multiple Infinispan operators exist at the global level, we need a way to distinguish between each operator's configuration.

Reply:
I see your point... how about using Infinispan CR API version?
Different operators must watch different API versions otherwise they clash, so something like:
`infinispan-operator-config-v1' `infinispan-operator-config-v2alpha1`
wdyt?

Reply:
> Different operators must watch different API versions otherwise they clash

This isn't the case with the proposed architecture. Multiple operators will watch the same CRD versions, however a field in the CRs determines which operator performs the reconciliation.

I think we need the name to contain the specific operator version.

Reply:
As version a bit long, we can introduce version in the ConfigMap filed and validates it on reconcile stage.
In this case we should bind watch to the specific ConfigMap template name, not exact value only

Reply:
Looks like the coexistence of multiple configs needs a proper design and since seems a little OT respect the grafana issue I would propose to move it in a separate issue/pr. wdyt?